Example/RealmResultsController-iOSTests/RealmQueueManagerTests.swift

```diff
@@ -0,0 +1,122 @@
+        let queue = dispatch_queue_create("THREAD 2", DISPATCH_QUEUE_SERIAL)
+        Threading.executeOnQueue(queue, sync: true) {
+            self.queueManager.addOperation {
+                self.thread2BlockHasBeenFired = true
+                sleep(2)
+            }
+        }
+    }
+    
+    func oneBlockFired() -> Bool {
+        let a = self.thread1BlockHasBeenFired || self.thread2BlockHasBeenFired
+        return a && !(self.thread1BlockHasBeenFired && self.thread2BlockHasBeenFired)
+    }
+    
+    override func spec() {
+        context("addOperation(withBlock:)") {
+            context("enqueued operations") {
+                let queueManager = RealmQueueManager(serial: false)
+                var block1Executed = false
+                var block2Executed = false
+                let block1 = {
+                    block1Executed = true
+                    sleep(blocksTimeout)
+                }
+                let block2 = {
+                    block2Executed = true
+                    sleep(blocksTimeout)
+                }
+                beforeEach {
+                    queueManager.addOperation(withBlock: block1)
+                    queueManager.addOperation(withBlock: block2)
+                }
+                afterEach {
+                    block1Executed = false
+                    block2Executed = false
+                }
+                it("should have executed the block 1") {
+                    expect(block1Executed).toEventually(beTruthy())
+                }
+                it("should have enqueued the execution of the block 2") {
+                    expect(block2Executed).toEventually(beFalsy())
+                }
+                it("should have executed the block 2 when reaching the 5 seconds timeout") {
+                    expect(block2Executed).toEventually(beTruthy(), timeout: 5)
+                }
+            }
+            context("serial operations") {
+                let queueManager = RealmQueueManager(serial: true)
+                var block1Executed = false
+                var block2Executed = false
+                let block1 = {
+                    block1Executed = true
+                    sleep(blocksTimeout)
+                }
+                let block2 = {
+                    block2Executed = true
+                    sleep(blocksTimeout)
+                }
+                beforeEach {
+                    queueManager.addOperation(withBlock: block1)
+                    queueManager.addOperation(withBlock: block2)
+                }
+                afterEach {
+                    block1Executed = false
+                    block2Executed = false
+                }
+                it("should have instantly executed the block 1") {
+                    expect(block1Executed).to(beTruthy())
+                }
+                it("should have NOT enqueued the execution of the block 2") {
+                    expect(block2Executed).to(beTruthy())
+                }
```

Comment:
Description confusion. Here what is happening is that addOperation executes the block synchronously, so the `it` is not reached until both blocks finish (after 4 seconds sleep). 
"should execute block 1 synchronously"
"should execute block 2 synchronously"